app/workers/articles/publish_worker.rb

```diff
@@ -0,0 +1,21 @@
+module Articles
+  class PublishWorker
+    include Sidekiq::Worker
+
+    sidekiq_options queue: :medium_priority, lock: :until_executing
+
+    def perform
+      # find published articles for which notifications were not set yet (so the context notifications were not created)
+      published_articles = Article.where("published = true AND published_at BETWEEN ? AND ?", 30.minutes.ago,
+                                         Time.current)
```

Comment:
nit: a shorter way to write this.

```suggestion
      published_articles =  Article.where(published: true, published_at: 30.minutes.ago..Time.current)
```